lib/active_encode/engine_adapters/ffmpeg_adapter.rb

```diff
@@ -34,7 +34,13 @@ def create(input_url, options = {})
         FileUtils.mkdir_p working_path("outputs", new_encode.id)
 
         # Extract technical metadata from input file
-        `#{MEDIAINFO_PATH} --Output=XML --LogFile=#{working_path("input_metadata", new_encode.id)} "#{input_url}"`
+        curl_option = if options && options[:headers]
+                        headers = options[:headers].map { |k, v| "#{k}: #{v}" }
+                        (["--File_curl=HttpHeader"] + headers).join(",")
+                      else
+                        ""
+                      end
+        `#{MEDIAINFO_PATH} '#{curl_option}' --Output=XML --LogFile=#{working_path("input_metadata", new_encode.id)} "#{input_url}"`
```

Comment:
Is it okay to leave this as `''` when there aren't any headers?  How about adding the single quotes to line 39 instead?
`"'#{(["--File_curl=HttpHeader"] + headers).join(",")}'"`

Reply:
I was trying to avoid a monstrous interpolation like that but I guess `yield_self` could work too.

Reply:
:+1: I'm not sure if I'm ever seen `yield_self`.  Seems really similar to `tap`